tests/test_models.py

```diff
@@ -319,7 +321,7 @@ def test_nested_boolean_criterion_dto_mapping(self):
         leaf_criterion = nested_boolean_criterion.match_all[0]
         assert isinstance(leaf_criterion, GrantaServerApiListsDtoRecordListSearchCriterion)
         assert leaf_criterion.name_contains == "A"
-        assert leaf_criterion.user_role == GrantaServerApiListsDtoUserRole.OWNER
+        assert leaf_criterion.user_role.value == GrantaServerApiListsDtoUserRole.OWNER.value
```

Comment:
Would

```suggestion
        assert leaf_criterion.user_role == GrantaServerApiListsDtoUserRole.OWNER
```

Not also work?

Reply:
Interesting question, I wasn't sure, but I've checked and no, it's not equivalent.
Members of different enums are not equal even if they have the same value.
The assertion could be:
```assert leaf_criterion.user_role == UserRole.OWNER```

I guess that's what you meant actually?

Reply:
Yes, I think so. Comparing the enum member itself, as opposed to the value of the member.

Reply:
I'm sorry you were right, I had misunderstood the test. We should expect the result to be `GrantaServerApiListsDtoUserRole.OWNER`. It is not and that's a bug. I've not updated the _to_model method to correctly instantiate an enum.